docs/validation.md

```diff
@@ -616,21 +616,220 @@ While this behaviour is unexpected (issues [#129](https://github.com/ajv-validat
 }
 ```
 
-The schema above is also more efficient - it will compile into a faster function.
+The schema above is also more efficient - it will compile into a faster function. Alternatively, you may use the [option `removeUnevaluated`](./validation.md#removing-unevaluated-properties).
 
 ### Removing unevaluated properties
 
-With [option `removeUnevaluated`](./api.md#options) you can filter data during the validation.
+With [option `removeUnevaluated`](./api.md#options) you can filter data during the validation. It is only compatible with draft 2019-09.
+
+This option modifies original data. It removes every property or item that would raise an error based on the keywords `unevaluatedProperties` and `unevaluatedItems`. It will first remove unevaluated properties then unevaluated items.
+
+Compared to `removeAdditional`, it will always take into account all sub-schemas that were valid for this data instance : 
+
+
+- all subschemas schemas in `allOf` and `$ref` keywords
+- valid sub-schemas in `oneOf` and `anyOf` keywords
+- sub-schema in `if` keyword
+- sub-schemas in `then` or `else` keywords that were applied based on the validation result by `if` keyword.
+
+
+
+
+Example 1 : remove unevaluated properties 
+
+```javascript
+
+const ajv = new Ajv({removeUnevaluated: true})
+
+const schema = {
+  $schema: "http://json-schema.org/draft/2019-09/schema#",
+  definitions: {
+    reference1: {
+      type: "object",
+      properties: { a: { type: "string" } }
+    },
+    reference2: {
+      type: "object",
+      properties: { a: { type: "string" } },
+      unevaluatedProperties: true
```

Comment:
is it expected that
schema.unevaluatedProperties = false
but mixed in with 
$ref.unevaluatedProperties = true

will be true?

Reply:
It is consistent with how ajv works but I will take a look at the spec to find the rationale.

Reply:
My understand is that the lowest unevaluatedProperties in the "tree" is the one that will be validated, so yes this is expected.